aws_lambda_powertools/utilities/idempotency/persistence/base.py

```diff
@@ -279,14 +279,14 @@ def _save_to_cache(self, data_record: DataRecord):
         -------
 
         """
-        if not self.use_local_cache:
+        if not self.use_local_cache or self._cache is None:
```

Comment:
That doesn't seem right. This is a function that saves it to the cache. By fixing Mypy hint, you're introducing a new bug. 

This line reads as: If customer enables cache BUT the cache is empty, just return.

Reply:
This validation checks if self_cache is None not empty dict. 
The dict is empty when you set `use_local_cache`, so  there is no chance of having use_local_cache variable set and self_cache set to None (https://github.com/awslabs/aws-lambda-powertools-python/blob/0196883410bf4fadeb1ca27f3f331ec6fbaffbe4/aws_lambda_powertools/utilities/idempotency/persistence/base.py#L157). Mypy doesn't know it and raise an issue that you cannot add dict key to None value (not dict). That's why I added this check to make it explicit. 
Anyway, if `_self_cache` was None code would fail with `TypeError: 'NoneType' object does not support item assignment` because we would try to add key to None value here (https://github.com/awslabs/aws-lambda-powertools-python/blob/0196883410bf4fadeb1ca27f3f331ec6fbaffbe4/aws_lambda_powertools/utilities/idempotency/persistence/base.py#L286) which is not correct.

Reply:
I see, lemme have a second look at this and I'll come back to you. We're doing this within `configure()` but MyPy isn't smart enough to check this

Reply:
Hmmm, I think mypy has this problem since we explicitly set this internal variable in init method:
```
self._cache: Optional[LRUDict] = None
```
Because of it mypy will not allow ungarded None in methods that use it.
https://mypy.readthedocs.io/en/stable/kinds_of_types.html#optional-types-and-the-none-type

Since we rely on `self.use_local_cache` that is always initialised to False and modified only when Configure() method is called with specific config AND we always check against it's value in every method that also would try to use _cache we can safely remove `self._cache: Optional[LRUDict] = None` from __init__ method.

This way mypy stops complaining about it and we don't need to add ignore in 4 places. WDYT @heitorlessa ?